src/com/pentalog/pentastagiu/homework2/shop/Customer.java

```diff
@@ -0,0 +1,51 @@
+package com.pentalog.pentastagiu.homework2.shop;
+
+import com.pentalog.pentastagiu.homework2.shop.customermembership.MembershipType;
+
+public class Customer {
+    private long id;
+    private String name;
+    private Basket basket;
+    private MembershipType membershipType;
+
+    public Customer(long id, String name, Basket basket, MembershipType membershipType) {
+        this.id = id;
```

Comment:
o sugestie de imbunatatire: pentru id sa ai tu un mecanism implementat pentru generarea lui, sa nu il lasi pe cel care creeaza obiectele sa ti-l paseze in constructor; in mecanismul tau poti controla de exemplu daca e unic si cam asta ne intereseaza cand avem un identificator pentru un obiect; un mecanism despre care am vb si in webinar e folosirea unui field static care sa fie incrementat de fiecare data cand se mai creeaza un obiect din clasa respectiva